clientapi/auth/storage/accounts/postgres/accounts_table.go

```diff
@@ -146,8 +146,12 @@ func (s *accountsStatements) selectAccountByLocalpart(
 }
 
 func (s *accountsStatements) selectNewNumericLocalpart(
-	ctx context.Context,
+	ctx context.Context, txn *sql.Tx,
 ) (id int64, err error) {
-	err = s.selectNewNumericLocalpartStmt.QueryRowContext(ctx).Scan(&id)
+	stmt := s.selectNewNumericLocalpartStmt
+	if txn != nil {
+		stmt = txn.Stmt(stmt)
```

Comment:
This is the sort of thing we used to use `common.TxStmt` for just so it was a bit more terse —
```
func TxStmt(transaction *sql.Tx, statement *sql.Stmt) *sql.Stmt {
	if transaction != nil {
		statement = transaction.Stmt(statement)
	}
	return statement
}
```

Reply:
I want to kill `common.TxStmt` at some point because in 99% of cases we should always be using the `txn` and never passing in `nil`. Making it terser makes it less clear what is going on IMO.

Reply:
In that case is it better if the reference code hard-fails if `txn` is `nil`?

Reply:
In this case I am explicitly passing in `nil` when people call the public function `GetNewNumericLocalpart(ctx context.Context) (int64, error)` - I want to kill this function too, but I also want to keep this PR on point which is why I'm doing the `nil` thing.

Reply:
No problem - let's merge this as-is then and maybe we can add a kind of assertion function later that checks `txn` is not `nil` and `panic()` if it is (that seems like a good excuse to panic as it would mean something is seriously wrong).

Reply:
SGTM!